python/demo/gsoc2022/demo_sbc.py

```diff
@@ -0,0 +1,244 @@
+# Copyright (C) 2022 Michele Castriotta
+#
+# ====================================================
+# Demo: scattering of an electromagnetic field by an
+#       infinite wire (scattering boundary conditions)
+# ====================================================
+
+from dolfinx import io, fem, plot, cpp
+from petsc4py import PETSc
+import ufl
+from ufl import FacetNormal, as_vector, inner, grad, curl, cross, lhs, rhs, dot, conj, sqrt
+from datetime import datetime
+import numpy as np
+import gmsh
+from mpi4py import MPI
+import os
+import sys
+from gmsh_helpers import gmsh_model_to_mesh
+from utils import background_electric_field, curl_2d, radial_distance, from_2d_to_3d, calculateAnalyticalEfficiencies, save_as_xdmf
+from scipy.constants import mu_0, epsilon_0
+
+# constant definition
+um = 10**-6  # micron
+nm = 10**-9  # nanometer
+pi = np.pi
+
+# radius of the wire and of the boundary of the domain
+radius_wire = 0.050 * um
+radius_dom = 1 * um
+
+# the smaller the mesh_factor, the finer is the mesh
+mesh_factor = 1.2
+
+# finite element degree
+degree = 3
+
+# wavelength sweep
+wl0 = 0.4 * um
+
+# background refractive index
+n_bkg = 1.33
+eps_bkg = n_bkg**2
+
+# mesh size inside the wire
+in_wire_size = mesh_factor * 7 * nm
+
+# mesh size at the boundary of the wire
+on_wire_size = mesh_factor * 3 * nm
+
+# mesh size in the vacuum
+bkg_size = mesh_factor * 60 * nm
+
+# mesh size at the boundary
+boundary_size = mesh_factor * 30 * nm
+
+# tags for the subdomains
+au_tag = 1          # gold wire
+bkg_tag = 2         # background
+boundary_tag = 3    # boundary
+
+# mesh definition in gmsh
+
+gmsh.initialize(sys.argv)
+if MPI.COMM_WORLD.rank == 0:
```

Comment:
Perhaps it's possible to split out the mesh generation into a different file?

Reply:
I have added a file named `mesh_wire.py` with a function for mesh generation.